src/elliptic/curves/test.rs

```diff
@@ -192,7 +192,7 @@ fn from_coords_produces_the_same_point<E: Curve>() {
     println!("s={}", s.to_bigint());
 
     let p: E::Point = <E::Point as ECPoint>::generator().scalar_mul(&s);
-    let coords = p.coords().unwrap();
+    let coords = p.coords();
```

Comment:
I think some new tests are necessary here. At least, that Scalar serialization/deserialization works for random and zero scalars. I can write tests